File: python/src/hywrapper/data/skyblock/essence_type.py
from __future__ import annotations

from enum import Enum
from typing import Any, Optional


class EssenceType(str, Enum):
    CRIMSON = "CRIMSON"
    DIAMOND = "DIAMOND"
    DRAGON = "DRAGON"
    FOREST = "FOREST"
    GOLD = "GOLD"
    ICE = "ICE"
    SPIDER = "SPIDER"
    UNDEAD = "UNDEAD"
    WITHER = "WITHER"

    @property
    def clean_name(self) -> str:
        return _CLEAN_NAMES[self]

    @classmethod
    def from_id(cls, id: Any) -> Optional[EssenceType]:
        try:
            return cls(id)
        except ValueError:
            return None


_CLEAN_NAMES = {
    EssenceType.CRIMSON: "Crimson",
    EssenceType.DIAMOND: "Diamond",
    EssenceType.DRAGON: "Dragon",
    EssenceType.FOREST: "Forest",
    EssenceType.GOLD: "Gold",
    EssenceType.ICE: "Ice",
    EssenceType.SPIDER: "Spider",
    EssenceType.UNDEAD: "Undead",
    EssenceType.WITHER: "Wither",
}

File: kotlin/src/main/kotlin/com/joshy/hywrapper/data/skyblock/EssenceType.kt
package com.joshy.hywrapper.data.skyblock

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class EssenceType(val cleanName: String) {
    @SerialName("CRIMSON")
    CRIMSON("Crimson"),

    @SerialName("DIAMOND")
    DIAMOND("Diamond"),

    @SerialName("DRAGON")
    DRAGON("Dragon"),

    @SerialName("FOREST")
    FOREST("Forest"),

    @SerialName("GOLD")
    GOLD("Gold"),

    @SerialName("ICE")
    ICE("Ice"),

    @SerialName("SPIDER")
    SPIDER("Spider"),

    @SerialName("UNDEAD")
    UNDEAD("Undead"),

    @SerialName("WITHER")
    WITHER("Wither"),
}
